src/pykiso/lib/auxiliaries/can_auxiliary.py

```diff
@@ -0,0 +1,209 @@
+        """
+        log.internal_info("Auxiliary instance deleted")
+        return True
+
+    def _receive_message(self, timeout_in_s: float = 0.1) -> None:
+        """Trigger connector reception method and put received messages
+        in the queue.
+
+        :param timeout_in_s: maximum time in second to wait for a response
+        """
+        try:
+            rcv_data = self.channel.cc_receive(timeout=timeout_in_s)
+            if rcv_data.get("msg") is not None:
+                log.internal_debug(f"received message '{rcv_data}' from {self.channel}")
+                message_name = self.parser.dbc.get_message_by_frame_id(rcv_data["remote_id"]).name
+                message_signals = self.parser.decode(rcv_data["msg"], rcv_data["remote_id"])
+                message_timestamp = float(rcv_data.get("timestamp", 0))
+                old_can_message = self.can_messages.get(message_name, None)
+                if old_can_message is None:
+                    can_msg = CanMessage(message_name, message_signals, message_timestamp)
+                    self.can_messages[message_name] = can_msg
+                else:
+                    self.can_messages[message_name].name = message_name
+                    self.can_messages[message_name].signals = message_signals
+                    self.can_messages[message_name].time_stamp = message_timestamp
+        except (Exception, KeyError):
+            log.exception(f"encountered error while receiving message via {self.channel}")
+            pass
+
+    def get_last_message(self, message_name: str) -> Optional[CanMessage]:
+        """Get the last message which has been received on the bus.
+        :param message_name: name of the message, you want to get
+
+        :return: last can massage or return none if the message, or return none if message is not occur 
+        """
+
+        return self.can_messages.get(message_name, None)
+
+
+    def get_last_signal(self, message_name: str, signal_name: str) -> Optional[Any]:
+        """Get the last message which has been received on the bus.
+        :param message_name: name of the message, you want to get
+
+        :return: last can massage or return none if the message or return none if message or signal is not occur
+        """
+
+        last_can_message = self.can_messages.get(message_name, None)
+        if last_can_message is not None:
+            return last_can_message.signals.get(signal_name, None)
+
+    def wait_for_message(
+        self,
+        message_name: str,
+        timeout: float = 0.2
+    ) -> dict[str, any]:
+        """Get the last message with certain timeout in seconds.
+        :param message_name: name of the message to receive
+        :param timeout time to wait till a message receives in seconds
+
+        :return: list of last can messages or None if no messages for this component
+        """
+        last_msg = self.can_messages.get(message_name, None)
+        time.sleep(timeout)
```

Comment:
This isn't precise at all, it anyone wants to know how long it took until the message arrived he can't

Instead of having the potential message as `can_messages` values, you could also have a Queue with maxsize=1 which would allow you to use the non-polling APIs `.get(timeout=...)`